### PR TITLE
render sign in/sign out in header based on auth state

### LINKED ISSUE #102: render sign in/sign out in header based on auth state

### PR DESCRIPTION
**Pull request by cezaryskura**
[ close #102 ]
- add event listener 'signedOut' and redirect to home
- collapse nav onclick
- change items name in nav
- set conditional rendering based on auth state in nav
- emit value when user signedOut
- change animation of nav open/close button

### CODE REVIEW

File: src/components/Nav.vue [hunk 1]
```diff
@@ -52,28 +53,40 @@
...
                 </router-link>
               </li>
 
               <li class="dropdown__menu-item py-3 ml-5">
                 <div class="dropdown__menu-link-text--small text-left">
-                  <router-link :to="routes.termsAndConditions"
-                    >Terms and conditions</router-link
-                  >
+                  <router-link :to="routes.termsAndConditions">
+                    <span>Terms and conditions</span>
```
Comment (IAfanasovMob) [inline src/components/Nav.vue:81]: does `span` changes anything here?

File: src/components/Nav.vue [hunk 2]
```diff
@@ -52,28 +53,40 @@
...
                 </div>
               </li>
-              <li class="dropdown__menu-item py-3 ml-5">
+              <li class="dropdown__menu-item pt-3 pb-5 ml-5">
                 <div class="dropdown__menu-link-text--small text-left">
-                  <router-link :to="routes.termsAndConditions"
-                    >Privacy statement</router-link
-                  >
+                  <router-link :to="routes.termsAndConditions">
+                    <span>Privacy statement</span>
```
Comment (IAfanasovMob) [inline src/components/Nav.vue:88]: and here

File: src/components/Nav.vue [hunk 3]
```diff
@@ -86,15 +99,55 @@
...
+    AmplifyEventBus.$on('authState', (state) => {
+      if (state === 'signedIn') {
+        this.isUserLoggedIn()
+      } else {
+        this.signedIn = false
+      }
+    })
+  },
+  methods: {
+    async isUserLoggedIn() {
```
Comment (IAfanasovMob) [inline src/components/Nav.vue:132]: name `isUserLoggedIn` suggest that it returns boolean

Comment (IAfanasovMob) [reply src/components/Nav.vue:132]: maybe `updateIsUserSignedIn`?

File: src/components/Nav.vue [hunk 4]
```diff
@@ -86,15 +99,55 @@
...
+      try {
+        await Auth.currentAuthenticatedUser()
+        this.signedIn = true
+      } catch (e) {
```
Comment (IAfanasovMob) [inline src/components/Nav.vue:136]: catching an exception is a kinda strange way to check if the user signed in. is it limitation of amplify?

File: src/components/Nav.vue [hunk 5]
```diff
@@ -86,15 +99,55 @@
...
+        this.signedIn = false
+      }
+    },
+    async signOut() {
+      this.showNav = false
+      try {
+        await Auth.signOut()
+        this.signedIn = false
```
Comment (IAfanasovMob) [inline src/components/Nav.vue:144]: if the user would be signed out somewhere else, the value of `this.signedIn` will not be updated. the event bus would be a great solution for such case

File: src/main.js [hunk 6]
```diff
@@ -44,8 +44,8 @@ Amplify.configure({
 })
 AmplifyEventBus.$on('authState', async (state) => {
   switch (state) {
-    case 'confirmSignUp':
-      router.push({ path: routes.entrepreneur })
+    case 'signedOut':
```
Comment (IAfanasovMob) [inline src/main.js:47]: I see state values used in several places already. worth to do a constant with a list of possible states

File: src/views/SignIn.vue [hunk 7]
```diff
@@ -19,15 +19,22 @@
         :required="true"
       />
 
+      <span
+        v-if="errors.includes(ErrorType.USER_NOT_FOUND_EXCEPTION)"
+        class="p-3 d-block text-danger"
+        >User does not exist. Please check your credentials.</span
+      >
+
       <button class="m-4 big-red-button" @click="signIn">Sign In</button>
```
Comment (IAfanasovMob) [inline src/views/SignIn.vue:28]: `type='button'`

File: src/components/Nav.vue (hunk 1 above)
Comment (cezaryskura) [reply src/components/Nav.vue:81]: not really.
I was thinking it will be semantically better. 
Span removed

File: src/components/Nav.vue (hunk 3 above)
Comment (cezaryskura) [reply src/components/Nav.vue:132]: done

File: src/views/SignIn.vue (hunk 7 above)
Comment (cezaryskura) [reply src/views/SignIn.vue:28]: done

File: src/components/Nav.vue (hunk 5 above)
Comment (cezaryskura) [reply src/components/Nav.vue:144]: Event bus from line 123 won't fix it ?

File: src/main.js (hunk 6 above)
Comment (cezaryskura) [reply src/main.js:47]: done

**Pull request merged**